diff -aru cmis-create-patch/old-src/libcmis/http-session.cxx cmis-create-patch/src/libcmis/http-session.cxx
--- cmis-create-patch/old-src/libcmis/http-session.cxx	2016-03-02 14:59:05.000000000 +0100
+++ cmis-create-patch/src/libcmis/http-session.cxx	2016-08-17 18:21:27.172963106 +0200
@@ -249,6 +249,8 @@
     curl_easy_setopt( m_curlHandle, CURLOPT_HEADERFUNCTION, &lcl_getHeaders );
     curl_easy_setopt( m_curlHandle, CURLOPT_WRITEHEADER, response.get() );
 
+    curl_easy_setopt( m_curlHandle, CURLOPT_VERBOSE, 1 );
+
     // fix Cloudoku too many redirects error
     // note: though curl doc says -1 is the default for MAXREDIRS, the error i got
     // said it was 0
@@ -304,6 +306,8 @@
 
     libcmis::HttpResponsePtr response( new libcmis::HttpResponse( ) );
 
+    curl_easy_setopt( m_curlHandle, CURLOPT_VERBOSE, 1 );
+
     curl_easy_setopt( m_curlHandle, CURLOPT_WRITEFUNCTION, lcl_bufferData );
     curl_easy_setopt( m_curlHandle, CURLOPT_WRITEDATA, response->getData( ).get( ) );
 
@@ -390,6 +394,8 @@
 
     libcmis::HttpResponsePtr response( new libcmis::HttpResponse( ) );
 
+    curl_easy_setopt( m_curlHandle, CURLOPT_VERBOSE, 1 );
+
     curl_easy_setopt( m_curlHandle, CURLOPT_WRITEFUNCTION, lcl_bufferData );
     curl_easy_setopt( m_curlHandle, CURLOPT_WRITEDATA, response->getData( ).get( ) );
 
@@ -474,6 +480,9 @@
     initProtocols( );
 
     curl_easy_setopt( m_curlHandle, CURLOPT_CUSTOMREQUEST, "DELETE" );
+
+    curl_easy_setopt( m_curlHandle, CURLOPT_VERBOSE, 1 );
+
     try
     {
         httpRunRequest( url );
Only in cmis-create-patch/src/libcmis: http-session.cxx~
